### PR TITLE
Fix/audio input

### LINKED ISSUE #259: AudioInput does not compile with Mozzi 2.0
(Initially reported on the forum)

Example `Audio_Input` does not compile with Mozzi 2.0. Looks like a simple typo, will provide a PR in a minute.

### PR DESCRIPTION
**Pull request by tomcombriat**
Fix #259

### CODE REVIEW

File: internal/config_checks_generic.h [hunk 1]
```diff
@@ -126,7 +126,7 @@ MOZZI_CHECK_POW2(MOZZI_CONTROL_RATE)
 #error "MOZZI_AUDIO_INPUT depends on MOZZI_ANALOG_READ option"
 #endif
 
-#if !MOZZI_IS(MOZZI_AUDIO_INPUT, MOZZI_AUDIO_INPUT_NONE) && defined(MOZZI_AUDIO_INPUT_PIN)
+#if MOZZI_IS(MOZZI_AUDIO_INPUT, MOZZI_AUDIO_INPUT_NONE) && defined(MOZZI_AUDIO_INPUT_PIN)
```
Comment (tfry-git) [inline internal/config_checks_generic.h:129]: This one was correct before: Shows a warning, if MOZZI_AUDIO_INPUT_PIN is defined without MOZZI_AUDIO_INPUT

**Review by tfry-git — APPROVED** [review]
I think the change in config_checks_generic.h is wrong, but otherwise: go ahead!

File: internal/config_checks_generic.h (hunk 1 above)
Comment (tomcombriat) [reply internal/config_checks_generic.h:129]: Are you sure?
 
The old one is: If Mozzi_input is Not None (hence, it is defined) and a pin is defined, shows an error. This gives a warning when compiling the Audio_Input example.

The new one is: If Mozzi_input is None and a pin is defined, shows an error.

Comment (tfry-git) [reply internal/config_checks_generic.h:129]: You are right!

**Pull request merged**